assets/js/blocks/product-categories/block.js

```diff
@@ -2,73 +2,138 @@
  * External dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { noop } from 'lodash';
-import { SelectControl } from '@wordpress/components';
+import classnames from 'classnames';
+import { Component, createRef, Fragment } from '@wordpress/element';
+import { IconButton } from '@wordpress/components';
+import { repeat } from 'lodash';
 import PropTypes from 'prop-types';
+import { withInstanceId } from '@wordpress/compose';
 
 /**
  * Internal dependencies
  */
 import { buildTermsTree } from './hierarchy';
 
-function getCategories( { hasEmpty, isDropdown, isHierarchical } ) {
+function getCategories( { hasEmpty, isHierarchical } ) {
 	const categories = wc_product_block_data.productCategories.filter(
 		( cat ) => hasEmpty || !! cat.count
 	);
-	return ! isDropdown && isHierarchical ?
+	return isHierarchical ?
 		buildTermsTree( categories ) :
 		categories;
 }
 
 /**
  * Component displaying the categories as dropdown or list.
  */
-const ProductCategoriesBlock = ( { attributes, isPreview = false } ) => {
-	const { hasCount, isDropdown } = attributes;
-	const categories = getCategories( attributes );
-	const parentKey = 'parent-' + categories[ 0 ].term_id;
+class ProductCategoriesBlock extends Component {
+	constructor() {
+		super( ...arguments );
+		this.select = createRef();
+		this.onNavigate = this.onNavigate.bind( this );
+		this.renderList = this.renderList.bind( this );
+		this.renderOptions = this.renderOptions.bind( this );
+	}
 
-	const renderList = ( items ) => (
-		<ul key={ parentKey }>
-			{ items.map( ( cat ) => {
-				const count = hasCount ? <span>({ cat.count })</span> : null;
-				return [
-					<li key={ cat.term_id }>
-						<a href={ isPreview ? null : cat.link }>{ cat.name }</a> { count } { /* eslint-disable-line */ }
-					</li>,
-					!! cat.children && !! cat.children.length && renderList( cat.children ),
-				];
-			} ) }
-		</ul>
-	);
+	onNavigate() {
+		const { isPreview = false } = this.props;
+		const url = this.select.current.value;
+		if ( 'false' === url ) {
+			return;
+		}
+		const home = wc_product_block_data.homeUrl;
 
-	return (
-		<div className="wc-block-product-categories">
-			{ isDropdown ? (
-				<SelectControl
-					label={ __( 'Select a category', 'woo-gutenberg-products-block' ) }
-					options={ categories.map( ( cat ) => ( {
-						label: hasCount ? `${ cat.name } (${ cat.count })` : cat.name,
-						value: cat.term_id,
-					} ) ) }
-					onChange={ noop }
-				/>
-			) : (
-				renderList( categories )
-			) }
-		</div>
-	);
-};
+		if ( ! isPreview && 0 === url.indexOf( home ) ) {
```

Comment:
Is there any case that `0 === url.indexOf( home )` will evaluate to false?

If there is, what do you think about rewriting this check as:
```suggestion
		if ( ! isPreview && url.startsWith( home ) ) {
```

Reply:
IE doesn't support `string.startsWith`. Why do you think we'd need to rewrite it? This check is make sure all links are internal, so `url` should always start with `home`.

Reply:
> IE doesn't support string.startsWith.

I thought transpiling the code with Babel would take care of IE11 compatibility, am I missing something? When testing it, I see strings get the `startsWith` method and it works just fine even in IE11:

![image](https://user-images.githubusercontent.com/3616980/60116729-e5df5800-9778-11e9-9107-4a4ec40afb59.png)

> Why do you think we'd need to rewrite it?

There is no need to change it, `indexOf` works too. It was just that I usually prefer ES6 methods because I think they make code easier to read for newcomers, but it's just a question of taste. :slightly_smiling_face: 

> This check is make sure all links are internal, so url should always start with home.

Right, I was curious if there is any case in which a category link could be external.

Reply:
weird, I was just going on with what MDN said in its support matrix, and made the assumption since it didn't add forEach to NodeList that it wouldn't do this either 🤷🏼‍♀️

>  if there is any case in which a category link could be external.

I don't _think_ that's possible, since we're linking off to WP-created taxonomy archives. I suppose a plugin could interject and do anything it wanted, but that seems like it would be really edge case.